### PR TITLE
Implemented GeoNames SFT and Converter

### PR DESCRIPTION
**Pull request by tandalesc**
Added GeoNames SFT and Converter. Tested on GeoNames gazetteer data successfully.

### CODE REVIEW

File: gm-data-geonames/src/main/resources/reference.conf~ [hunk 1]
```diff
@@ -0,0 +1,62 @@
+geomesa = {
```
Comment (jnh5y) [inline gm-data-geonames/src/main/resources/reference.conf~]: Unnecessary file.  Watch out during commits!

**Comment by jnh5y** [comment]
Can you add a README.md for this config?  It should 1) link to the file it can ingest, and 2) explain what the data is.

File: gm-data-geonames/src/main/resources/reference.conf~ (hunk 1 above)
Comment (tandalesc) [inline gm-data-geonames/src/main/resources/reference.conf~]: removed!

**Comment by tandalesc** [comment]
@jnh5y I added a README.md file in /gm-data-geonames

File: gm-data-geonames/src/main/resources/reference.conf [hunk 2]
```diff
@@ -0,0 +1,56 @@
+geomesa = {
+	sfts = {
```
Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: use 2 spaces for indent in this file

Comment (jnh5y) [inline gm-data-geonames/src/main/resources/reference.conf]: @vpipkt @elahrvivaz Do we have a preference for spaces over tabs in converter config?

Any ideas about enforcing silly style things like that?

Comment (elahrvivaz) [inline gm-data-geonames/src/main/resources/reference.conf]: set it in intellij? but yeah, spaces FTW

File: gm-data-geonames/src/main/resources/reference.conf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			]
+		}
+	},
+	converters = {
+		geonames = {
+			type = "delimited-text",
+			format = "MYSQL",
+			id-field = "toString($geonameid)",
+			fields = [
+				{ name = "geonameid", 			transform = "try($1::int, null)" },
```
Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: picky, perhaps move the transforms back left a little when doing spaces

Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: and optionally line up all the closing brackets...(if you don't want to I won't die) :angel: sometimes i get crazy on the formatting desires

File: gm-data-geonames/src/main/resources/reference.conf [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+				{ name = "cc2", type = "String", index = false },
+				{ name = "admin1code", type = "String", index = false },
+				{ name = "admin2code", type = "String", index = false },
+				{ name = "admin3code", type = "String", index = false },
+				{ name = "admin4code", type = "String", index = false },
+				{ name = "population", type = "Long", index = false },
+				{ name = "elevation", type = "Integer", index = false },
+				{ name = "dem", type = "Integer", index = false },
+				{ name = "timezone", type = "String", index = false },
+				{ name = "modificationdate", type = "Date", index = false },
```
Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: consider camelCase or underscores? what do we do elsewhere?

File: gm-data-geonames/src/main/resources/reference.conf [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+				{ name = "geom", type = "Point", index = true, srid = 4326, default = true }
+			]
+		}
+	},
+	converters = {
+		geonames = {
+			type = "delimited-text",
+			format = "MYSQL",
```
Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: [picky] add small comment on this line indicating that the data is tsv but in MYSQL format

File: gm-data-geonames/src/main/resources/reference.conf [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+			id-field = "toString($geonameid)",
```
Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: what if this is null? perhaps line 33 should not allow geonameid to be null with the try and instead just blow out

File: gm-data-geonames/src/main/resources/reference.conf (hunk 3 above)
Comment (jnh5y) [inline gm-data-geonames/src/main/resources/reference.conf]: Seriously, there's got to be a tool which'll do this formatting stuff...

Comment (jahhulbert-ccri) [inline gm-data-geonames/src/main/resources/reference.conf]: we write an intellij plugin and force committers to use it

File: gm-data-geonames/src/main/resources/reference.conf (hunk 4 above)
Comment (elahrvivaz) [inline gm-data-geonames/src/main/resources/reference.conf]: i'd go with camelCase

File: gm-data-geonames/src/main/resources/reference.conf (hunk 6 above)
Comment (tandalesc) [inline gm-data-geonames/src/main/resources/reference.conf]: done!

File: gm-data-geonames/src/main/resources/reference.conf (hunk 5 above)
Comment (tandalesc) [inline gm-data-geonames/src/main/resources/reference.conf]: done!

File: gm-data-geonames/src/main/resources/reference.conf (hunk 4 above)
Comment (tandalesc) [inline gm-data-geonames/src/main/resources/reference.conf]: gm-data's .conf files are all mixed, some are camel case, some are all lowercase. I think geomesa's official docs use all lower case

File: gm-data-geonames/src/main/resources/reference.conf (hunk 3 above)
Comment (tandalesc) [inline gm-data-geonames/src/main/resources/reference.conf]: done and done!

File: gm-data-geonames/src/main/resources/reference.conf (hunk 2 above)
Comment (tandalesc) [inline gm-data-geonames/src/main/resources/reference.conf]: done!

File: gm-data-geonames/src/main/resources/reference.conf (hunk 4 above)
Comment (elahrvivaz) [inline gm-data-geonames/src/main/resources/reference.conf]: i'd say try to standardize on camel-case going forward (e.g. here :)

**Comment by jahhulbert-ccri** [comment]
@jnh5y for fun can help @tandalesc you have this done as a squashed single commit with the original commit having the "merge message" and name the branch something like "geonames" rather than master on his fork?

**Comment by jahhulbert-ccri** [comment]
after that it looks good and i'll merge

**Pull request closed without merging**